### DISCUSSION TITLE
Link Synchronization

### DISCUSSION THREAD

**Discussion by rlourenco-centi** [openthread/openthread#8817] [Q&A] [answered]
I was simulating some cases in your otns simulator , however I can't make any case where the Link synchronization is a 3 message exchange. I can only make an exchange of 2 messages. Could you explain to me how I can get to a 3 message exchange as is the case represented in the link - https://openthread.io/guides/thread-primer/router-selection

**Comment by jwhui** [comment, marked as answer]
A 3-way Link Request/Accept exchange occurs when the initial Link Request message is sent as a multicast.

One scenario where this occurs is when a router receives an MLE Advertisement from a neighbor that it has not yet set up a link with.

**Reply by rlourenco-centi** [reply]
I can't simulate this scenario in otns. When you said "has not yet set up a link with" you refer that node already is a child, right?
I would like to simulate the Link request in multicast in your simulator and I am not succeeding.
Can you send me a **otns_0.replay** file that corresponds  this scenario. 

Thank you,
Rafael.

**Reply by rlourenco-centi** [reply]
in the specification it is said that Link synchronization is performed when:
	- Attaching a parent - Unicast (2 message exchange but in the spec it said that is multicast request)
	- REED becames Router - It this scenario it can be multicast(3 messages) , right ? 
	- when a router recive a message that dont have stored the frame counter (Unicast - 2 message)
	- After hardware reset (Unicast - 2 message)

**Reply by jwhui** [reply]
Yes, when REED becomes Router, it will send a multicast MLE Link Request.

But 3-way exchange can happen via unicast as well, when the two routers do not have links established to each other.

**Reply by rlourenco-centi** [reply]
Another question about this topic. 
The link synchronization is always done by Routers in the same partition? if yes,a new child FTD can change the partition ID from this process(Link Synchronization) ? 
Can you help me in this issue too - https://github.com/openthread/ot-ns/issues/497 .

**Reply by jwhui** [reply]
> The link synchronization is always done by Routers in the same partition?

Yes, Thread devices only establish links to neighboring devices in the same partition.

> If yes,a new child FTD can change the partition ID from this process(Link Synchronization) ?

No, the Leader is responsible for establishing the Partition ID.
